idaes/apps/grid_integration/tests/test_perfectforecaster.py

```diff
@@ -0,0 +1,121 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+
+import pytest
+from pyomo.common import unittest as pyo_unittest
+from idaes.apps.grid_integration.forecaster import PerfectForecaster
+import idaes.logger as idaeslog
+import pandas as pd
+import numpy as np
+
+@pytest.fixture
+def wind_df():
+    start_year = 2020
+    start_mon = 1
+    start_day = 1
+    start_date = pd.Timestamp(f"{start_year}-{start_mon:02d}-{start_day:02d} 00:00:00")
+    ix = pd.date_range(start=start_date, 
+                        end=start_date
+                        + pd.offsets.DateOffset(days=1)
+                        - pd.offsets.DateOffset(hours=1),
+                        freq='1H')
+    df = pd.DataFrame(index = ix)
+    df["303_WIND_1-DACF"] = list(i/100 for i in range(24))
+    df["303_WIND_1-RTCF"] = list((i+1)/100 for i in range(24))
+    df["Caesar-DALMP"] = list(i for i in range(24))
+    df["Caesar-RTLMP"] = list(i+1 for i in range(24))
+    return df
+
+@pytest.fixture
+def base_perfectforecaster(wind_df):
+    return PerfectForecaster(wind_df)
+
+@pytest.mark.unit
+def test_create_perfectforecaster(wind_df):
+    perfectforecaster = PerfectForecaster(data_path_or_df=wind_df)
+    assert perfectforecaster.data is wind_df
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("value", [np.array([1,2,3,4,5]), [1,2,3,4,5]])
+def test_create_perfectforecaster_with_ndarray_and_list(value):
+    with pytest.raises(ValueError):

```

Comment:
When checking for errors, you should also include the expected text message to ensure you got the right error (and not another of the same type).